nautobot/core/tests/test_api.py

```diff
@@ -15,3 +17,9 @@ def test_status(self):
         response = self.client.get("{}?format=api".format(url), **self.header)
 
         self.assertEqual(response.status_code, 200)
+
+    def test_non_existent_resource(self):
+        url = reverse("api-root")
+        response = self.client.get(f"{url}/non-existent-resource-url/", **self.header)
```

Comment:
I'd expect to also see a `self.assertEqual(response.status_code, 404)` here, I think?